### PR TITLE
Fix #81, #83: Explain how to uses toolchain files and define missing Windows Exports

### LINKED ISSUE #81: Documment how to use toolchain files with the FindPackage
### Issue information

|                       Field | Value |
|----------------------------:|-------|
|          **Jira issue key** | `INSTALL-490` |
|                   **OS(s)** | `--` |
|      **Connext version(s)** | `--` |
| **Connext architecture(s)** | `--` |

### Description

Explain and show an example of a toolchain file of a supported architecture (e.g.: QNX).

### LINKED ISSUE #83: Export the services specific Windows symbol export defines for the new services (CDS, WIS and PS)
### Issue information

|                       Field | Value |
|----------------------------:|-------|
|          **Jira issue key** | `INSTALL-792` |
|                   **OS(s)** | `Windows` |
|      **Connext version(s)** | `--` |
| **Connext architecture(s)** | `--` |

### PR DESCRIPTION
**Pull request by lulivi**
Fixes #81
Fixes #83 

### Logs

<details>
  <summary>Before the fix</summary>
  <pre>
Done Building Project "C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\ALL_BUILD.vcxproj" (default targets) -- FAILED.

Build FAILED.

"C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\CloudDiscoveryServiceLibraryAPIC.vcxproj" (default target) (3) ->
(Link target) ->
  cloud_discovery_service_library_api.obj : error LNK2019: unresolved external symbol RTI_CDS_LOG_VERBOSITY_ALL referenced in function main [C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\CloudDiscoveryServiceLibraryAPIC.vcxproj]
  cloud_discovery_service_library_api.obj : error LNK2019: unresolved external symbol RTI_CDS_LOG_VERBOSITY_EXCEPTIONS referenced in function main [C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\CloudDiscoveryServiceLibraryAPIC.vcxproj]
  C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\Release\CloudDiscoveryServiceLibraryAPIC.exe : fatal error LNK1120: 2 unresolved externals [C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\CloudDiscoveryServiceLibraryAPIC.vcxproj]

    0 Warning(s)
    3 Error(s)

Time Elapsed 00:00:01.82
  </pre>
</details>

<details>
  <summary>After the fix</summary>
  <pre>
Done Building Project "C:\Users\testadmin\rticonnextdds-examples\examples\cloud_discovery_service\library_api\c\build\ALL_BUILD.vcxproj" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.24
  </pre>
</details>

### CODE REVIEW

**Review by fgaranda — APPROVED** [review]

**Review by fgaranda — APPROVED** [review]

File: cmake/Modules/FindRTIConnextDDS.cmake [hunk 1]
```diff
@@ -426,8 +426,38 @@
 # Other platforms and architectures might work, but they are not oficially
 # supported.
 #
+# Toolchain examples
+# ^^^^^^^^^^^^^^^^^^
```
Comment (rkorte) [inline cmake/Modules/FindRTIConnextDDS.cmake:430]: Is this really a subsection of "Supported platforms"?

File: cmake/Modules/FindRTIConnextDDS.cmake [hunk 2]
```diff
@@ -350,13 +350,13 @@
 #   ``CONNEXTDDS_IMPORTED_TARGETS_DEBUG`` will not have any effect.
 #
 # Note
-# ^^^^
+# ----
 # Some flags related to the compiler, like (-std=c++11 needed when linking
```
Comment (rkorte) [inline cmake/Modules/FindRTIConnextDDS.cmake:354]: I think you mean this:

Some flags related to the compiler, (such as -std=c++11, needed when linking

File: cmake/Modules/FindRTIConnextDDS.cmake [hunk 3]
```diff
@@ -426,8 +426,38 @@
...
+#     set(CMAKE_LINKER "${CMAKE_C_COMPILER}" CACHE PATH "")
+#     set(CMAKE_AR "${tools_dir}/ntoaarch64-ar" CACHE PATH "")
+#
+#     set(PLATFORM_MODERN_CXX_STANDARD 11)
+#     set(CMAKE_CXX_EXTENSIONS TRUE)
+#
 # Logging in versions lower than CMake 3.15
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# -----------------------------------------
 # For versions lower than CMake 3.15, ``CONNEXTDDS_LOG_LEVEL`` variable should
```
Comment (rkorte) [inline cmake/Modules/FindRTIConnextDDS.cmake:461]: insert the word "the" before ``CONNEXTDDS_LOG_LEVEL``

File: cmake/Modules/FindRTIConnextDDS.cmake [hunk 4]
```diff
@@ -11,12 +11,12 @@
...
-# -----------------
+# =================
 #
 # Find RTI Connext DDS libraries.
 #
 # Components
-# ^^^^^^^^^^
+# ----------
 # This module sets variables for the following components that are part of RTI
 # Connext DDS:
```
Comment (rkorte) [inline cmake/Modules/FindRTIConnextDDS.cmake:21]: Just say RTI Connext (no DDS)

(don't bother searching and replacing all ... we should update this as we can, and be consistent within a file ... if changing all instances in a file is too big a job, then leave it alone)

**Review by rkorte — CHANGES_REQUESTED** [review]
comments indicated

**Review by fgaranda — DISMISSED** [review]

**Review by rkorte — APPROVED** [review]

**Pull request merged**